AppConnectSwift/CaptureImageViewController.swift

```diff
@@ -48,12 +90,17 @@ class CaptureImageViewController: UIViewController, UIImagePickerControllerDeleg
     
     @IBAction func saveTapped(sender: AnyObject) {
         if image.CGImage != nil {
-            let data = compressFile() as? NSData
-            var subject: MDSubject!
-            subject.collectData(data, withMetadata: "Random String", completion: { (dataEnvelope: MDSubjectDataEnvelope!, err: NSError!) -> Void in
-                print(err == nil ? "Data Saved" : err?.description);
-                self.imageView.image = nil
-            })
+            for subject in self.collectedSubjects {
+                subject.collectData(self.data, withMetadata: "Random String", completion: { (dataEnvelope: MDSubjectDataEnvelope!, err: NSError!) -> Void in
+                    NSOperationQueue.mainQueue().addOperationWithBlock {
+                        if err == nil {
+                            print(err == nil ? "Data Saved" : err?.description)
+                        }
```

Comment:
I think this ternary should be pulled out of the conditional? Or else one of the conditions will never be reached.